esmvalcore/preprocessor/_area.py

```diff
@@ -243,33 +255,36 @@ def area_statistics(cube, operator, fx_files=None):
     ValueError
         if input data cube has different shape than grid area weights
     """
-    grid_areas = tile_grid_areas(cube, fx_files)
+    if operator == 'weightless_sum':
+        return cube.collapsed(['longitude', 'latitude'], iris.analysis.SUM)
+
+    if operator == 'weightless_mean':
+        return cube.collapsed(['longitude', 'latitude'], iris.analysis.MEAN)
+
+    operation = get_iris_analysis_operation(operator)
+
+    # Many IRIS analysis functions do not accept weights arguments.
+    if not operator_accept_weights(operator):
+        return cube.collapsed(['longitude', 'latitude'], operation)
 
     if not fx_files and cube.coord('latitude').points.ndim == 2:
         logger.error(
             'fx_file needed to calculate grid cell area for irregular grids.')
         raise iris.exceptions.CoordinateMultiDimError(cube.coord('latitude'))
 
-    coord_names = ['longitude', 'latitude']
-    if grid_areas is None or not grid_areas.any():
-        cube = guess_bounds(cube, coord_names)
+    if not calculate_grid:
+        grid_areas = tile_grid_areas(cube, fx_files)
+    else:
+        cube = guess_bounds(cube, ['longitude', 'latitude'])
```

Comment:
check if previous bounds exist

Reply:
`guess_bounds` does this check already.

Reply:
no it don't, it fails if there are previous bounds

Reply:
```
# guess bounds tool
def guess_bounds(cube, coords):
    """Guess bounds of a cube, or not."""
    # check for bounds just in case
    for coord in coords:
        if not cube.coord(coord).has_bounds():
            cube.coord(coord).guess_bounds()
    return cube
```

Reply:
ah yes, that's the wrapper of guess bounds from iris, my bad (I put the wrapper in and I forget about it)